app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -136,14 +141,85 @@ public View inflateLayout(LayoutInflater inflater, @Nullable ViewGroup container
 
         progressView = (ProgressBar) view.findViewById(R.id.progress);
 
-        view.findViewById(R.id.erase).setOnClickListener(this);
-
         menuView = view.findViewById(R.id.menu);
         menuView.setOnClickListener(this);
 
+        if (BrowsingSession.getInstance().isCustomTab()) {
+            initialiseCustomTabUi(view);
+        } else {
+            initialiseNormalBrowserUi(view);
+        }
+
         return view;
     }
 
+    private void initialiseNormalBrowserUi(final @NonNull View view) {
+        final View erase = view.findViewById(R.id.erase);
+        erase.setOnClickListener(this);
+
+        urlView.setOnClickListener(this);
+    }
+
+    private void initialiseCustomTabUi(final @NonNull View view) {
+        final CustomTabConfig customTabConfig = BrowsingSession.getInstance().getCustomTabConfig();
+        if (customTabConfig == null) {
+            throw new IllegalStateException("Can't initialise custom tab UI for non custom-tab session");
+        }
+
+        // Unfortunately there's no simpler way to have the FAB only in normal-browser mode.
+        // - ViewStub: requires splitting attributes for the FAB between the ViewStub, and actual FAB layout file.
+        //             Moreover, the layout behaviour just doesn't work unless you set it programatically.
+        // - View.GONE: doesn't work because the layout-behaviour makes the FAB visible again when scrolling.
+        // - Adding at runtime: works, but then we need to use a separate layout file (and you need
+        //   to set some attributes programatically, same as ViewStub).
+        final View erase = view.findViewById(R.id.erase);
+        final ViewGroup eraseContainer = (ViewGroup) erase.getParent();
+        eraseContainer.removeView(erase);
```

Comment:
Maybe we could extend the FloatingActionButton class to permanently hide the button in some states.